src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,280 @@
+        Finds the orthogonal transformation that best aligns the estimated rotations
+        to the reference rotations.
+
+        :param rots_ref: The reference Rotation object to which we would like to align
+            with data matrices in the form of a n-by-3-by-3 array.
+        :return: o_mat, optimal orthogonal 3x3 matrix to align the two sets;
+                flag, flag==1 then J conjugacy is required and 0 is not.
+        """
+        rots = self.data
+        rots_ref = rots_ref.rot_matrices.astype(self.dtype)
+        ensure(
+            rots.shape == rots_ref.shape,
+            "Two sets of rotations must have same dimensions.",
+        )
+        K = rots.shape[0]
+
+        # Reflection matrix
+        J = np.array([[1, 0, 0], [0, 1, 0], [0, 0, -1]])
+
+        Q1 = np.zeros((3, 3), dtype=rots.dtype)
+        Q2 = np.zeros((3, 3), dtype=rots.dtype)
+
+        for k in range(K):
+            R = rots[k, :, :]
+            Rref = rots_ref[k, :, :]
+            Q1 = Q1 + R @ Rref.T
+            Q2 = Q2 + (J @ R @ J) @ Rref.T
+
+        # Compute the two possible orthogonal matrices which register the
+        # estimated rotations to the true ones.
+        Q1 = Q1 / K
+        Q2 = Q2 / K
+
+        # We are registering one set of rotations (the estimated ones) to
+        # another set of rotations (the true ones). Thus, the transformation
+        # matrix between the two sets of rotations should be orthogonal. This
+        # matrix is either Q1 if we recover the non-reflected solution, or Q2,
+        # if we got the reflected one. In any case, one of them should be
+        # orthogonal.
+
+        err1 = norm(Q1 @ Q1.T - np.eye(3), ord="fro")
+        err2 = norm(Q2 @ Q2.T - np.eye(3), ord="fro")
+
+        # In any case, enforce the registering matrix O to be a rotation.
+        if err1 < err2:
+            # Use Q1 as the registering matrix
+            U, _, V = svd(Q1)
+            flag = 0
+        else:
+            # Use Q2 as the registering matrix
+            U, _, V = svd(Q2)
+            flag = 1
+
+        Q_mat = U @ V
+
+        return Q_mat, flag
+
+    def get_aligned_rotations(self, Q_mat, flag):
+        """
+        Get aligned Rotation object to reference ones.
+
+        Calculated aligned rotation matrices from the orthogonal transformation
+        that best aligns the estimated rotations to the reference rotations.
+
+        :param Q_mat:  optimal orthogonal 3x3 transformation matrix
+        :param flag:  flag==1 then J conjugacy is required and 0 is not
+        :return: regrot, aligned Rotation object
+        """
+        rots = self.data
+        K = rots.shape[0]
+
+        # Reflection matrix
+        J = np.array([[1, 0, 0], [0, 1, 0], [0, 0, -1]])
+
+        regrot = np.zeros_like(rots)
+        for k in range(K):
+            R = rots[k, :, :]
+            if flag == 1:
+                R = J @ R @ J
+            regrot[k, :, :] = Q_mat.T @ R
+        aligned_rots = copy.copy(self)
```

Comment:
Why copy? Just create a new `Rotation` object.

Reply:
Changed to return a new `Rotation` object in commit e863d01.